avm/ptn/app-service-lza/hosting-environment/modules/naming/naming.module.bicep

```diff
@@ -0,0 +1,2265 @@
+/**
+ * Azure naming module - helps maintaining a consistent naming convention
+ * Licensed to use under the MIT license.
+ * ----------------------------------------------------------------------------
+ * Module repository & documentation: https://github.com/nianton/azure-naming
+ * Starter repository template:       https://github.com/nianton/bicep-starter
+ * ----------------------------------------------------------------------------
+ * Microsoft naming convention best practices (supports user-defined types and compile time imports)
+ * https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming
+ * ----------------------------------------------------------------------------
+ * Generated/built on: 2024-05-02T13:30:03.233Z
+ */
+
+metadata name = 'Azure Naming module'
+metadata description = 'Module to maintain a consistent naming of Azure resources.'
+metadata owner = 'https://github.com/nianton'
```

Comment:
Please note that this entire module will be loaded into the deployment even though you only use a fraction of its content. Also it may make sense to publish it as its own module.

Reply:
Yeah I'm going to strip it down now that you pointed that out. I do think we need a naming module for use in AVM pattern modules. This is an open source naming convention I used so I didn't write it so not sure how that works with publishing it as it's own module.

Reply:
That's fair - and I won't ask you to either :) It's quite unfortuante that Bicep has a 4MB limit for deployments, making modules for things like 'naming' not impossible, but quite ... delicate 😄